src/core/session/codec.go

```diff
@@ -0,0 +1,66 @@
+// Copyright Project Harbor Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package session
+
+import (
+	"encoding/gob"
+
+	"github.com/astaxie/beego/session"
+	commonmodels "github.com/goharbor/harbor/src/common/models"
+	"github.com/goharbor/harbor/src/lib/cache"
+	"github.com/goharbor/harbor/src/lib/errors"
+)
+
+func init() {
+	gob.Register(commonmodels.User{})
+}
+
+var (
+	// codec the default codec for the cache
+	codec cache.Codec = &gobCodec{}
+)
+
+type gobCodec struct{}
```

Comment:
Because there is a cache codec, maybe you could reuse this one
https://github.com/goharbor/harbor/blob/a1d37384a179cec27c2c5ca86a9f7167a06ca2e2/src/lib/cache/codec.go#L32

Reply:
We need to implement the new codec(align with beego implementation) for compatibility with regression and upgrade.